src/processor_cxx17/msg_handler.cxx

```diff
@@ -0,0 +1,154 @@
+#include "msg_handler.hxx"
+
+#include "control_message.hxx"
+#include "control_message_ack.hxx"
+#include "control_message_ack_with_checksum.hxx"
+#include "control_message_nack.hxx"
+#include "data_message.hxx"
+
+#include <arpa/inet.h>
+#include <cstring>
+#include <iostream>
+#include <variant>
+#include <vector>
+
+namespace cpp17_app {
+
+MessageHandler::MessageHandler(boost::asio::io_context &io_context):
+    net::TcpServerConnection(io_context)
+{
+}
+
+void MessageHandler::cbReceive(const std::vector<std::byte>& data)
+{
+    if (!data.size()) {
+        std::cerr << __func__ << ": buffer is empty" << std::endl;
+        return;
+    }
+
+    // this place should be optimized with PMR
+    container_t raw_msg(data);
+    std::unique_ptr<MessageVariant> msg_variant;
```

Comment:
this ugly unique_ptr, I really prefer to use std::optional for the case when you do not want to construct the object immediately